projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/graphql/GraphQLDataFetchers.java

```diff
@@ -227,6 +236,49 @@ private List<V2DataJob> populateDeployments(
     return allDataJob;
   }
 
+  private Map<String, JobDeploymentStatus> readJobDeploymentsFromK8s() {
+    return deploymentService.readDeployments().stream()
+        .collect(Collectors.toMap(JobDeploymentStatus::getDataJobName, cronJob -> cronJob));
+  }
+
+  private Map<String, JobDeploymentStatus> readJobDeploymentsFromDb() {
+    var deployments =
+        StreamSupport.stream(actualJobDeploymentRepository.findAll().spliterator(), false)
+            .collect(Collectors.toMap(ActualDataJobDeployment::getDataJobName, cronjob -> cronjob));
+
+    return deployments.entrySet().stream()
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey, entry -> convertToJobDeploymentStatus(entry.getValue())));
+  }
+
+  private JobDeploymentStatus convertToJobDeploymentStatus(
+      ActualDataJobDeployment deploymentStatus) {
+    JobDeploymentStatus jobDeploymentStatus = new JobDeploymentStatus();
+    jobDeploymentStatus.setDataJobName(deploymentStatus.getDataJobName());
+    jobDeploymentStatus.setPythonVersion(deploymentStatus.getPythonVersion());
+    jobDeploymentStatus.setGitCommitSha(deploymentStatus.getGitCommitSha());
+    jobDeploymentStatus.setEnabled(deploymentStatus.getEnabled());
+    jobDeploymentStatus.setLastDeployedBy(deploymentStatus.getLastDeployedBy());
+    jobDeploymentStatus.setLastDeployedDate(deploymentStatus.getLastDeployedDate().toString());
+    jobDeploymentStatus.setResources(getDataJobResources(deploymentStatus.getResources()));
+    // The ActualDataJobDeployment does not have a mode attribute, which is required by the
+    // JobDeploymentStatus,
+    // so we need to set something in order to avoid errors.
+    jobDeploymentStatus.setMode("release");
+
+    return jobDeploymentStatus;
+  }
+
+  private DataJobResources getDataJobResources(DataJobDeploymentResources deploymentResources) {
+    DataJobResources resources = new DataJobResources();
```

Comment:
null check

Reply:
Did a complete refactoring and re-used an already existing method.